### ISSUE TITLE
String method contains()

### ISSUE THREAD

**Issue by trych** [basiljs/basil.js#331]
One issue that always comes up in my classes is the need to check if a string contains a string.

While we have the methods `startsWith()` and `endsWith()`, we're missing a method to check if a string is contained in a string.

I suggest we add a method `contains()`.

**Comment by fabianmoronzirfas** [comment]
Yes it should. Can be done with `indexOf`:

```js
var basil = "basil";
if(basil.indexOf("sil") !== -1 ){
        $.writeln("found sil")
    }

if(basil.indexOf("foo") !== -1 ){
        $.writeln("found foo")
    }

```

**Comment by trych** [comment]
Yes, perfect. And should we call it `contains()` or `includes()`. I would prefer `contains()`, so it would not get mixed up with the JS ES6 `.includes()`. But if you guys prefer otherwise, I am open for alternatives.

**Comment by fabianmoronzirfas** [comment]
+1 for contains
There is also a array contains function in es5 could be cool to make this work with strings and arrays

**Comment by fabianmoronzirfas** [comment]
I was wrong it is also `Array.includes`[polyfill link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#Polyfill). Still `contains(thing: array|string|object /*non recursive*/): boolean` would be cool

**Comment by trych** [comment]
So you mean one single function that works for all these data types, right?

**Comment by fabianmoronzirfas** [comment]
Maybe. 🤔 Don't you think it would be useful ? Objects would be with hasOwnProperty

**Comment by trych** [comment]
Array would be useful for sure, object probably once in a while. But sure, no harm in including them.

**Comment by fabianmoronzirfas** [comment]
Hm. I think object is ambiguous. Should it look for the `key` or the `value`? 
And actually is a shallow check not of much use. So maybe we leave object out.

**Comment by trych** [comment]
Yes, sounds better somehow.

**Comment by trych** [comment]
Also, startsWith and endsWith could then also be changed to work for arrays as well. Not sure, if anyone ever needs that, but at least all three methods would work consistently then. 😛

**Comment by trych** [comment]
Implemented via #332 .
